src/server/database.py

```diff
@@ -25,38 +25,26 @@ def get_user_id(user_dict):
     return db.users.find_one({"mac": user_dict["mac"]})
 
 
-def get_users_by_os(os):
-    return db.users.find({"os": os})
-
-
-def get_copied_phrases(n=100):
-    pipeline = [
-        {"$match": {"copy_pastaed": True}},
-        {"$group": {"_id":"$phrase", "num": {"$sum":1}}},
-        {"$limit": int(n)},
-        {"$project": {"_id": 0, "phrase": "$_id", "num_occurences":"$num"}}
-    ]
-    return db.phrases.aggregate(pipeline)
-
-
 def get_or_create_user(user):
     user_dict = user.__dict__
-    user_obj = get_user_id(user_dict)
-    if user_obj is None:
+    if get_user_id(user_dict) is None:
+        user_dict['phrases'] = []
         db.users.insert_one(user_dict)
     return get_user_id(user_dict)
 
 
 def insert_phrases(user, phrases_list):
-    user_id = get_user_id(user.__dict__)
-    if user_id is None:
+    if get_user_id(user.__dict__) is None:
         logger.error("insert_phrases: Bad user id")
-    update_user(user)
+        return None
+    # User has updated tags, so add phrases that the user has typed
+    user_id = user.__dict__
     for p in phrases_list:
         phrase = p.__dict__
-        phrase['user_id'] = str(user_id)
+        user_id['phrases'].append(phrase)
         db.phrases.insert_one(phrase)
```

Comment:
So then it's debatable whether we need a phrases collection or not. I say we should keep it for now, but at the same time we can also remove it as otherwise we just do an unwind, and then a group by some phrase attribute.

Reply:
If we are putting phrases in users, then we have no need for the phrases collection and should remove it. We don't need to store phrases twice.